ls_pre_commit_hooks/plantuml-renderer.sh

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+if ! command -v plantuml &>/dev/null; then
+  if command -v brew &>/dev/null; then
+    brew install plantuml
+  else
+    >&2 echo 'plantuml command not found. See https://plantuml.com/download for installation instructions.'
+    exit 1
+  fi
+fi
+
+for puml_file in "${@}"; do
+  plantuml -tsvg "${@}" -o "images"
+done
+git add \*.svg
```

Comment:
I think it should not do a git add. All the other pre-commit hooks we have just update the files without staging them in git: java code formatter, black python formatter, detect-secrets, whitespace trimmer/end-of-file checker, etc. they all either modify the file or check if it's correct but never `git add` anything directly.

But the proper way to do it would be: `git add "${puml_file%.*}.svg"` inside the loop. This will replace the `.puml` with `.svg` and add exactly the svg for the files that were changed.

See https://stackoverflow.com/a/965069

Reply:
I thought about making it fail, but I am not sure how to make it fail.
Any suggestions.? Do you think by checking git stage status is sufficient? Since its svg file, if no change then there should be no update at all.

Reply:
I think it should be enough to run `plantuml`. I think pre-commit will detect by itself if there is any file changed then it will make it fail so that you can't commit unless you git add both the plantuml file and the svg file.

Reply:
No, it doesnt fail if the script successfully executed. And it will just commit with unstaged file.

Reply:
I think I wont fail it, it is okay to just git add it and it will only run at local too.
I don't want to complexify it, since to make it properly work as a "checker" I will need to put it at tmp dir, check the difference and so on..

Also the pre-commit name is "renderer" anyway not "checker".. 😛